E0391 needs to be updated to new format

From: src/test/compile-fail/issue-20825.rs

Error E0391 needs a span_label, updating it from:

```
error[E0391]: unsupported cyclic reference between types/traits detected
  --> src/test/compile-fail/issue-20825.rs:15:41
   |
15 | pub trait Processor: Subscriber<Input = Self::Input> {
   |                                         ^^^^^^^^^^^
   |
   = note: the cycle begins when computing the supertraits of `Processor`...
   = note: ...which then again requires computing the supertraits of `Processor`, completing the cycle.
```

To:

```
error[E0391]: unsupported cyclic reference between types/traits detected
  --> src/test/compile-fail/issue-20825.rs:15:41
   |
15 | pub trait Processor: Subscriber<Input = Self::Input> {
   |                                         ^^^^^^^^^^^ cyclic reference
   |
   = note: the cycle begins when computing the supertraits of `Processor`...
   = note: ...which then again requires computing the supertraits of `Processor`, completing the cycle.
```

--- a/src/librustc_resolve/lib.rs
+++ b/src/librustc_resolve/lib.rs
@@ -219,7 +219,13 @@ fn resolve_struct_error<'b, 'a: 'b, 'c>(resolver: &'b Resolver<'a>,
                              name)
         }
         ResolutionError::IsNotATrait(name) => {
-            struct_span_err!(resolver.session, span, E0404, "`{}` is not a trait", name)
+            let mut err = struct_span_err!(resolver.session,
+                                           span,
+                                           E0404,
+                                           "`{}` is not a trait",
+                                           name);
+            err.span_label(span, &format!("not a trait"));
+            err
         }
         ResolutionError::UndeclaredTraitName(name, candidates) => {
             let mut err = struct_span_err!(resolver.session,

--- a/src/librustc_typeck/collect.rs
+++ b/src/librustc_typeck/collect.rs
@@ -184,6 +184,7 @@ impl<'a,'tcx> CrateCtxt<'a,'tcx> {
 
         let mut err = struct_span_err!(tcx.sess, span, E0391,
             "unsupported cyclic reference between types/traits detected");
+        err.span_label(span, &format!("cyclic reference"));
 
         match cycle[0] {
             AstConvRequest::GetItemTypeScheme(def_id) |

--- a/src/test/compile-fail/issue-3907.rs
+++ b/src/test/compile-fail/issue-3907.rs
@@ -18,6 +18,7 @@ struct S {
 }
 
 impl Foo for S { //~ ERROR: `Foo` is not a trait
+                 //~| NOTE: not a trait
                  //~| NOTE: type aliases cannot be used for traits
     fn bar() { }
 }

--- a/src/test/compile-fail/issue-5035.rs
+++ b/src/test/compile-fail/issue-5035.rs
@@ -11,6 +11,7 @@
 trait I {}
 type K = I;
 impl K for isize {} //~ ERROR: `K` is not a trait
+                    //~| NOTE: not a trait
                     //~| NOTE: aliases cannot be used for traits
 
 use ImportError; //~ ERROR unresolved

--- a/src/test/ui/codemap_tests/two_files.stderr
+++ b/src/test/ui/codemap_tests/two_files.stderr
@@ -2,7 +2,7 @@ error[E0404]: `Bar` is not a trait
   --> $DIR/two_files.rs:16:6
    |
 16 | impl Bar for Baz { }
-   |      ^^^
+   |      ^^^ not a trait
    |
    = note: type aliases cannot be used for traits
 

